From 5aff7fcd6437b7c8af7294c761e15c2c925bdfda Mon Sep 17 00:00:00 2001
From: Peter Swain <peter.swain@cavium.com>
Date: Wed, 22 Aug 2018 14:14:54 -0700
Subject: [PATCH 0143/1051] misc: cavium-tdm: remove trace_printk calls

disable all trace_printk() unless CONFIG_CAVIUM_TDM_DEBUG=y

Signed-off-by: Peter Swain <peter.swain@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/misc/cavium-tdm.c | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/drivers/misc/cavium-tdm.c b/drivers/misc/cavium-tdm.c
index 0634901f0cfc..afcdd4394bb0 100644
--- a/drivers/misc/cavium-tdm.c
+++ b/drivers/misc/cavium-tdm.c
@@ -101,12 +101,13 @@ enum {
 static int tdm_trace = t_init;
 module_param_named(trace, tdm_trace, int, 0644);
 
+/* event tracing, which collapses to nothing if !defined(DEBUG) */
+#ifdef CONFIG_CAVIUM_TDM_DEBUG
 /* trace_printk logging of interesting events (slot map changes) */
 static int old_trace; /* saved trace flags over change-tracing */
 static int trace_changes; /* count down on exhaustive logging */
 
-/* event tracing, which collapses to nothing if !defined(DEBUG) */
-#define tr_tdm(mask, fmt, ...) do { \
+# define tr_tdm(mask, fmt, ...) do { \
 	/* unless trace_changes, trace only initial N */ \
 	static int _once = 50; \
 	/* TODO: adjust masks on WARN calls, rather than this strstr() */ \
@@ -120,6 +121,12 @@ static int trace_changes; /* count down on exhaustive logging */
 		} \
 	    } \
 	} while (0)
+#else /*!_DEBUG*/
+# define tr_tdm(mask, fmt, ...) do { \
+		if (tdm_trace & (mask)) \
+			pr_debug(DEVNAME " " fmt, ##__VA_ARGS__);\
+	} while (0)
+#endif /*!_DEBUG*/
 
 /*
  * A bus is one or more engines wired-or'd, currently exactly one,
@@ -696,6 +703,7 @@ static struct tdm_bus veng[NR_TDM_ENGINES];
 /* caller holds lock, mark an "interesting" event */
 static inline void mark_change(void)
 {
+#ifdef CONFIG_CAVIUM_TDM_DEBUG
 	bool was_off = !old_trace;
 
 	if (!(tdm_trace & t_change))
@@ -707,11 +715,13 @@ static inline void mark_change(void)
 		trace_printk("tdm tracing on %x |= %x\n",
 			tdm_trace, t_detail);
 	tdm_trace |= t_detail;
+#endif
 }
 
 /* caller holds lock, mark irq-handling done */
 static inline void mark_done(void)
 {
+#ifdef CONFIG_CAVIUM_TDM_DEBUG
 	if (trace_changes <= 0)
 		return;
 	if (--trace_changes)
@@ -719,6 +729,7 @@ static inline void mark_done(void)
 	trace_printk("tdm tracing off\n");
 	tdm_trace = old_trace;
 	old_trace = 0;
+#endif
 }
 
 static unsigned long held;
-- 
2.17.1

